File: kotlin/Earth/Earth.kt
package Earth

interface ContinentType {
    val name: String;
    val area: Double;
    val population: Number;
}

class Continent(val name: String, val area: Double, val population: Number)

val radius: Double = 6.4e3;

val continents = listOf<Continent>(
        Continent("Africa",30.3e6, 1.216e9 ),
        Continent("Antartica",13e6, 4.0e3 ),
        Continent("Asia",43.3e6, 4.561e9 ),
        Continent("Australia",7.7e6, 25.69e8 ),
        Continent("Europa",10.4e6, 746.4e8 ),
        Continent("North America",24.9e6, 579e8 ),
        Continent("South America",17.8e6, 422.5e8 )
)

File: rust/earth/src/earth.rs
pub struct Continent {
	pub name: String,
	pub area: f64,
	pub population: f64
}

pub fn continentList() -> [Continent; 7] {
	return [
		Continent {
			name: "Africa".to_string(),
			area: 32.3e6,
			population: 1.216e9
		},
		Continent {
			name: "Antartica".to_string(),
			area: 13e6,
			population: 4.0e3
		},
		Continent {
			name: "Asia".to_string(),
			area: 43.3e6,
			population: 4.561e9
		},
		Continent {
			name: "Australia".to_string(),
			area: 7.7e6,
			population: 25.69e8
		},
		Continent {
			name: "Europa".to_string(),
			area: 10.4e6,
			population: 746.4e8
		},
		Continent {
			name: "North America".to_string(),
			area: 24.9e6,
			population: 579e8
		},
		Continent {
			name: "South America".to_string(),
			area: 17.8e6,
			population: 422.5e8
		},
	]
}

pub const RADIUS: f32 = 6.4e3;